src/foraging_gui/Foraging.py

```diff
@@ -205,8 +205,9 @@ def connectSignalsSlots(self):
         self.StartBleaching.clicked.connect(self._StartBleaching)
         self.NextBlock.clicked.connect(self._NextBlock)
         self.OptogeneticsB.activated.connect(self._OptogeneticsB) # turn on/off optogenetics
-        self.OptogeneticsB.currentIndexChanged.connect(self._keyPressEvent)
-        self.PhotometryB.currentIndexChanged.connect(self._keyPressEvent)
+        self.OptogeneticsB.currentIndexChanged.connect(lambda: self._QComboBoxUpdate('Optogenetics',self.OptogeneticsB.currentText()))
+        self.PhotometryB.currentIndexChanged.connect(lambda: self._QComboBoxUpdate('Photometry',self.PhotometryB.currentText()))
+        self.FIPMode.currentIndexChanged.connect(lambda: self._QComboBoxUpdate('FIPMode', self.FIPMode.currentText()))
```

Comment:
We may log the change for all `QComboBox` in a more general way. It doesn’t have to be done here.

        for container in [self.TrainingParameters, self.centralwidget, self.Opto_dialog]:
            # Iterate over each child of the container that is a QLineEdit or QDoubleSpinBox
            for child in container.findChildren((QtWidgets.QComboBox)):        
                child.currentIndexChanged.connect(lambda: self._QComboBoxUpdate())

Reply:
Well, we dont want to do it for all QComboboxes, because some of them we map to specific functions. I agree we should find a better solution

Reply:
I think one change can be mapped to multiple functions in parallel.